cmd/xdpcap/xdpaction.go

```diff
@@ -15,8 +15,22 @@ const (
 	xdpDrop
 	xdpPass
 	xdpTx
+	xdpRedirect
+
+	// Magic action that means every XDP Action, even ones we may not know about yet
+	xdpAll xdpAction = -1
```

Comment:
Could you instead handle this at the flag parsing stage, and specify that the string "all" results in all know actions being returned? Then the rest of the code doesn't have to know about all, and weird cases like `all,drop` don't exist.

Reply:
`xdpAll` means all actions exposed by the hook map, not all the actions we know about. We can handle "all" specially for the flag parsing, but then we need to somehow pass that to `filter`. I can add a separate `allActions bool` to `filterOpts` if that's nicer, but we still have to deal with cases were `actions` contains something, and `allActions` is set.

Reply:
Turns out we can just get rid of `xdpAll`, giving filter an empty `filter.actions` now means use all the actions exposed by the map. I think that works quite nicely, having an empty set of actions doesn't make sense anyways.